google/src/main/scala/org/broadinstitute/dsde/rawls/google/GoogleUtilities.scala

```diff
@@ -36,11 +36,22 @@ trait GoogleUtilities extends LazyLogging with Retry {
     }
   }
 
-  protected def retryWhen500orGoogleError[T](op: () => T) = {
+  protected def whenGoogle404: Predicate[Throwable] = throwable => {
+    throwable match {
+      case t: GoogleJsonResponseException if t.getStatusCode == 404 => true
+      case _ => false
+    }
+  }
+
+  protected def retryWhen[T](pred: Predicate[Throwable])(op: () => T): Future[T] = {
```

Comment:
I'm also thinking of adding a spec for the `Retry` logic, since the logic tripped me up and I don't see any existing tests for it.